presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -883,11 +883,22 @@ private void writeRoleGrantsFile(Set<RoleGrant> roleGrants)
 
         List<String> partitionNames = partitions.stream()
                 .map(partitionValues -> makePartitionName(table.getPartitionColumns(), ImmutableList.copyOf(partitionValues)))
+                .filter(partitionName -> isValidPartition(table, partitionName))
```

Comment:
I'd squash `Fix FileHiveMetastore to check if partition ...` commit with the next commit.